### ISSUE TITLE
Build a program from compiled scripts

### ISSUE THREAD

**Issue by monokot** [paulbartrum/jurassic#173]
Dear sirs,

In my application, JavaScript is divided into modules that link to the final program. Is there any way to build a program from compiled scripts?  Otherwise (when changing the code of any of the modules), I have to combine the code of all the modules and recompile it. This procedure is very expensive, maybe there are some workarounds?

Thanks.

**Comment by paulbartrum** [comment]
Maybe I'm missing something, but can't you just run each compiled script against the same `ScriptEngine`, one after the other? There should be no need to combine scripts.

**Comment by paulbartrum** [comment]
I still don't really understand.

Here's what I'm suggesting:
1. Create new script engine.
2. Execute Module1.js. Script engine now contains compiled `module1_function` function.
3. Execute Module2.js. Previously compiled `module1_function` function is called.

Now you're left with a ScriptEngine with everything that was defined in Module1.js and Module2.js. If Module2.js changes you can create a new script engine and run (previously compiled) Module1 and (newly compiled) Module2 against that new script engine.

**Comment by monokot** [comment]
Yes, this is what I need, thank you very much for the detailed description of the actions. I didn't pay attention to the Execute method of the CompiledScript class.
